dbus/src/blocking.rs

```diff
@@ -204,20 +211,46 @@ impl $c {
         self.remove_match_no_cb(&mr.match_str())
     }
 
+    /// If true, configures the connection to send signal messages to all matching [`MatchRule`]
+    /// filters added with [`add_match`](Self::add_match) rather than just the first one. This will
+    /// result in the messages being duplicated, and so the message serial will be lost, but this is
+    /// generally not a problem for signals.
+    ///
+    /// This is false by default, for a newly-created connection.
+    pub fn all_signal_matches(&mut self, all_signal_matches: bool) {
+        self.all_signal_matches = all_signal_matches;
+    }
+
     /// Tries to handle an incoming message if there is one. If there isn't one,
     /// it will wait up to timeout
     ///
     /// This method only takes "&self" instead of "&mut self", but it is a logic error to call
     /// it recursively and might lead to panics or deadlocks.
     pub fn process(&self, timeout: Duration) -> Result<bool, Error> {
         if let Some(msg) = self.channel.blocking_pop_message(timeout)? {
-            let ff = self.filters_mut().remove_matching(&msg);
-            if let Some(mut ff) = ff {
-                if ff.2(msg, self) {
-                    self.filters_mut().insert(ff);
+            if self.all_signal_matches && msg.msg_type() == MessageType::Signal {
+                // If it's a signal and the mode is enabled, send a copy of the message to all
+                // matching filters.
+                for mut ff in self.filters_mut().remove_all_matching(&msg) {
+                    if let Ok(copy) = msg.duplicate() {
```

Comment:
That all matching signals are removed during the callback process is a bit...interesting. It means that 1) if one callback panics, the remaining ones will be dropped and 2) one callback can remove earlier matches in the chain but not later ones, which is potentially confusing. 

That said; as long as this is properly documented as gotchas with the `all_signal_matches` mode, I guess that could be okay for the time being - if it confuses someone else they can write a patch to improve the situation.

Reply:
Perhaps something like "when the all_signal_match mode is enabled, removing other matches inside a match callback is not supported".

Reply:
Added to the docstring for `set_all_signal_matches`, is there anywhere else you think I should mention it?